sys/include/riotboot/slot.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_riotboot_slot   Helpers to manipulate partitions (slots)
+ *                              on internal flash
+ * @ingroup     sys
+ * @{
+ *
+ * @file
+ * @brief       Slot management tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#ifndef RIOTBOOT_SLOT_H
+#define RIOTBOOT_SLOT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "riotboot/hdr.h"
+
+/**
+ * @brief  Get currently running image slot
+ *
+ * @returns nr of currently active slot
+ */
+int riotboot_slot_current(void);
+
+/**
+ * @brief  Get jump-to address of image slot
+ *
+ * @param[in]   slot    slot nr to work on
+ *
+ * @returns address of first byte of @p slot
+ */
+uint32_t riotboot_slot_get_image_startaddr(unsigned slot);
+
+/**
+ * @brief  Boot into image in slot @p slot
+ *
+ * @param[in]   slot    slot nr to jump to
+ */
+void riotboot_slot_jump(unsigned slot);
+
+/**
+ * @brief  Get header from a given flash slot
+ *
+ * @param[in]   slot    slot nr to work on
+ *
+ * @returns header of image slot nr @p slot
+ */
+const riotboot_hdr_t *riotboot_slot_get_hdr(unsigned slot);
+
+/**
+ * @brief  Validate slot
+ *
+ * @param[in] slot    slot nr to work on
+ *
+ * @returns 0 if ok.
+ */
+static inline int riotboot_slot_validate(unsigned slot)
```

Comment:
The goal is to avoid any explicit dependency of the modules using `riotboot_slot` from `riotboot_hdr`. Encapsulating here this function avoids to include `riotboot_hdr.h`.

The other function only uses code belonging to this module, so no need to encapsulate it in any way.

Reply:
> Encapsulating here this function avoids to include riotboot_hdr.h.

You can keep `#include riotboot/hdr.h` in `slot.h`, move the function definitions to `slot.c`, and keep only the declarations in `slot.h`. The user wouldn't need to explicitly include `riotboot_hdr.h` in that case either. And you wouldn't have function definitions in header files, which is generally bad practice.